keylime/tpm/tpm_util.py

```diff
@@ -236,6 +234,14 @@ def checkquote(
     if isinstance(pubkey, EllipticCurvePublicKey) and sig_alg not in [tpm2_objects.TPM_ALG_ECDSA]:
         raise ValueError(f"Unsupported quote signature algorithm '{sig_alg:#x}' for EC keys")
 
+    if sig_alg in [tpm2_objects.TPM_ALG_RSASSA]:
+        (sig_size,) = struct.unpack_from(">H", sigblob, 4)
+        (signature,) = struct.unpack_from(f"{sig_size}s", sigblob, 6)
+    elif sig_alg in [tpm2_objects.TPM_ALG_ECDSA]:
+        signature = ecdsa_der_from_tpm(sigblob)
```

Comment:
This is probably the better function to call. The commit text does not say anything, but did this not work (correctly) so far? Presumably `pubkey.verify()` of the ECDSA key would have thrown an error... ?

Reply:
Yes `pubkey.verify()` has thrown an error at least on non P256 curves.

Reply:
Since swtpm uses P384 I thought that it had been working at least with that one...

Reply:
https://github.com/keylime/rust-keylime/pull/846 actually implements non default key sizes/curves. Before I started working on this the necessary support in the rust binding for generating the correct EKs was missing.